serverless.yml

```diff
@@ -22,52 +23,53 @@ provider:
     FAULTLINE_USE_KMS: ${env:FAULTLINE_USE_KMS, file(./config.yml):useKms, '0'}
     FAULTLINE_KMS_KEY_ALIAS: ${env:FAULTLINE_KMS_KEY_ALIAS, file(./config.yml):kmsKeyAlias, 'faultline'}
     FAULTLINE_VERSION: ${file(./package.json):version}
-  runtime: nodejs12.x
+  runtime: nodejs16.x
   stage: ${self:provider.environment.FAULTLINE_STAGE}
   region: ${self:provider.environment.FAULTLINE_REGION}
-  deploymentBucket: ${self:provider.environment.FAULTLINE_DEPLOYMENT_BUCKET}
```

Comment:
Is it possible not to drop `deploymentBucket:` if I set it up as shown in the following URL?

ref: https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml#deployment-bucket

Reply:
@k1LoW
If `deploymentBucket.name` is written with an empty string (it is default value), the configuration file validation will detect the following warning:

```
Warning: Invalid configuration encountered
  at 'provider.deploymentBucket.name': must NOT have fewer than 3 characters
  at 'provider.deploymentBucket.name': must match pattern "(?!^(\d{1,3}\.){3}\d{1,3}$)(^(([a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9])$)"
```

This is a validation-only issue, and the internal code of the Serverless Framework seems to treat `deploymentBucket.name` as null if it is an empty string.

https://github.com/serverless/serverless/blob/bb37f4fe75ff5234fae48ada433cd52ddf51cb91/lib/plugins/aws/provider.js#L221-L232
https://github.com/serverless/serverless/blob/bb37f4fe75ff5234fae48ada433cd52ddf51cb91/lib/plugins/aws/provider.js#L898-L919

To avoid this warning, users need to specify the `deploymentBucket`, because S3 bucket name is should be globally unique, the default value cannot be set.

I have determined that there are few use cases where `deploymentBucket` must be explicitly specified, so I removed the configuration value.

Reply:
How about comment outed `deploymentBucket` section in severless.yml?

```yaml
## Uncomment this section, if you want to specify deployment bucket.
#  deploymentBucket:
#    name: ${env:FAULTLINE_DEPLOYMENT_BUCKET, file(./config.yml):deploymentBucket}
```

Reply:
> How about comment outed deploymentBucket section in severless.yml?

Looks Good!!

Reply:
@k1LoW Fixed!